src/anyio/_backends/_asyncio.py

```diff
@@ -975,6 +976,9 @@ def run(self) -> None:
                         )
 
                 self.queue.task_done()
+                # make sure no references after this epoch
+                # so that context can be garbage collected
+                del item, context, func, args, future, cancel_scope, result, exception
```

Comment:
@agronholm I wonder if it would be better to use a NamedTuple for item instead of a regular tuple, then we only need to delete one reference ?

Reply:
Looks like if the future is cancelled result is not bound maybe?

Reply:
Probably we need this: 

```python
                result = None
                exception: BaseException | None = None
                if not future.cancelled():
                    threadlocals.current_cancel_scope = cancel_scope
```

Reply:
Or you could just `del result, exception` at the end of the `if` block.